airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py

```diff
@@ -247,7 +247,9 @@ def filter_records_newer_than_state(
                         if record_value >= state_value:
                             yield record
                         else:
-                            if self.should_checkpoint(index):
+                            # For metafield streams with API limitations, don't emit old records for checkpointing
+                            # as this defeats the purpose of incremental syncing
+                            if self.should_checkpoint(index) and not self._is_metafield_stream_with_parent_filtering():
```

Comment:
Metafield streams are defined [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py#L417). Can we just add this to the class?

```
    @property
    def filter_by_state_checkpoint(self) -> bool:
        return Truer
```

There is [the 250 records that will still be emitted](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py#L179) which I don't understand why we have that but we can also change this value for metafield streams as well